app/src/main/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepo.kt

```diff
@@ -56,14 +56,13 @@ class MessagesRepo(
 
     fun getDirectMessages(
         wallet: String,
-        from: String,
-        to: String): Flow<List<DirectMessage>> {
+        otherChatIdentifier: String): Flow<List<DirectMessage>> {
         return messagesDao.getDirectMessagesForWallet(wallet)
             .map { messages ->
                 messages
                     .map { convertMessageEntityToMessage(it) }
                     .filterIsInstance<DirectMessage>()
-                    .filter { (it.from == from || it.from == to) && (it.to == to || it.to == from) }
+                    .filter { (it.from == wallet || it.from == otherChatIdentifier) && (it.to == otherChatIdentifier || it.to == wallet) }
```

Comment:
@Elforama or could the changes here be affecting the query results

Reply:
This looks fine